stm32-modules/heater-shaker/firmware/motor_task/motor_policy.cpp

```diff
@@ -99,3 +102,11 @@ auto MotorPolicy::plate_lock_set_power(float power) -> void {
 auto MotorPolicy::plate_lock_disable() -> void {
     motor_hardware_plate_lock_off(&hw_handles->tim3);
 }
+
+auto MotorPolicy::set_pid_constants(double kp, double ki, double kd) -> void {
+    // These conversions match those in drive_parameters.h and therefore let you just look
+    // at the numeric literals there
+    PID_SetKD(hw_handles->mct[0]->pPIDSpeed, static_cast<int16_t>(kd / (SPEED_UNIT / 10)));
```

Comment:
Why is "/ 10" needed? What is 10? Could it be folded into SPEED_UNIT or another named constant?

Reply:
It matches the logic here: https://github.com/Opentrons/opentrons-modules/blob/652bbbc848feceba61801114bbc078f6cf341e66/stm32-modules/heater-shaker/firmware/motor_task/drive_parameters.h#L95-L100

That logic meanwhile is there because of autogenerated code that almost but does not quite adapt itself for changing the SPEED_UNIT define. It's supposed to be a thing you can change between RPM, Hz, and 0.1Hz and it changes all the math; but the autogenerated code that works with it in fact slaps those factors-of-10 in there by default which only works for 0.1Hz and I really don't want to deal with changing it.

Because this is a test gcode handler, the benefit of having the factor-of-10 magic is that the pid constants the caller sets now match numerically the values that are set by default in the above link.

Reply:
Got it. Thanks for the explanation.